lnd_test.go

```diff
@@ -5386,13 +5387,151 @@ func testInvoiceSubscriptions(net *lntest.NetworkHarness, t *harnessTest) {
 	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPoint, false)
 }
 
-// testBasicChannelCreation test multiple channel opening and closing.
-func testBasicChannelCreation(net *lntest.NetworkHarness, t *harnessTest) {
+// channelSubscription houses the proxied update and error chans for a node's
+// channel subscriptions.
+type channelSubscription struct {
+	updateChan chan *lnrpc.ChannelUpdate
+	errChan    chan error
+	quit       chan struct{}
+}
+
+// subscribeChannelNotifications subscribes to channel updates and launches a
+// goroutine that forwards these to the returned channel.
+func subscribeChannelNotifications(ctxb context.Context, t *harnessTest,
+	node *lntest.HarnessNode) channelSubscription {
+
+	// We'll first start by establishing a notification client which will
+	// send us notifications upon channels becoming active, inactive or
+	// closed.
+	req := &lnrpc.ChannelSubscription{}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	chanUpdateClient, err := node.SubscribeChannels(ctx, req)
+	if err != nil {
+		t.Fatalf("unable to create channel update client: %v", err)
+	}
+
+	// We'll launch a goroutine that will be responsible for proxying all
+	// notifications recv'd from the client into the channel below.
+	errChan := make(chan error, 1)
+	quit := make(chan struct{})
+	chanUpdates := make(chan *lnrpc.ChannelUpdate, 20)
+	go func() {
+		for {
+			defer cancelFunc()
```

Comment:
I don't think the `cancelFunc` should be in the loop?

Reply:
fixed -- also an issue [here](https://github.com/lightningnetwork/lnd/blob/master/lnd_test.go#L7696). Can open a pr!